keps/sig-instrumentation/3077-contextual-logging/README.md

```diff
@@ -1037,6 +1079,11 @@ Revert commits that changed log calls.
 
 ## Implementation History
 
+* Kubernetes 1.24: initial alpha
```

Comment:
I can't comment on not modified text, so a few questions.  `Feature Enablement and Rollback` and `Version Skew Strategy` questions, both of these are gaining in importance when the feature is enabled by default. 
1. What will happen if a user starts to use new flags/options which start logging to json and rolls the component back? Are there any new flags for setting up different log output at all? 
2.  You mentioned in the risk section `Performance overhead risk` how a cluster administrator can notice that? Was this measured as part of this effort to point out which components are affected more and which less with that impact? You mentioned flat 2% but in the risk section you called out that there will be places where this might be more or less, that's why I'm asking about per-component data. 
3. `Pod scheduling (= "startup latency of schedulable stateless pods" SLI) might become slightly worse` - have you measured it by how much? This is similar to the above question. 
4.  Failure modes, in the risks section you described `Uninitialized logger`, is there a way we can prevent that failure from happening in CI? 
5.  In `What steps should be taken if SLOs are not being met to determine the problem?` I doubt the cluster operator can revert commits, you need to define this more in terms of a cluster admin rollback to previous working version.

Reply:
> What will happen if a user starts to use new flags/options which start logging to json and rolls the component back? 

JSON output is not part of this feature. That's [structured logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging).

> Are there any new flags for setting up different log output at all?

This feature here has no separate command line flags or config options.

> Was this measured as part of this effort to point out which components are affected more and which less with that impact?

We only have performance benchmarks for kube-scheduler. See https://github.com/kubernetes/enhancements/pull/4219#issuecomment-1807811995 for an evaluation of that.

> have you measured it by how much? This is similar to the above question.

I observed no relevant difference in the kube-scheduler benchmark.

> Failure modes, in the risks section you described Uninitialized logger, is there a way we can prevent that failure from happening in CI?

This is mostly a non-issue because the Go API was designed such that all code which asks for a `logr.Logger` gets one. A `Logger` value is never uninitialized. A pointer might be nil, but this is rarely used.

To catch mistakes in the CI, the modified code must be executed.

> I doubt the cluster operator can revert commits, you need to define this more in terms of a cluster admin rollback to previous working version.

True. Let me double-check what I wrote there.